src/backend/tools/compass.py

```diff
@@ -0,0 +1,153 @@
+import os
+import logging
+from typing import Any, Dict, List
+from compass_sdk import CompassDocument
+from compass_sdk import ParserConfig, MetadataConfig
+from compass_sdk.parser import CompassParserClient
+from compass_sdk.compass import CompassClient
+from backend.tools.base import BaseTool
+
+logger = logging.getLogger()
+
+class CompassTool(BaseTool):
+    """Tool to interact with a Compass instance."""
+    @classmethod
+    def is_available(cls) -> bool:
+        vars = [
+            "COHERE_COMPASS_URL",
+            "COHERE_COMPASS_USERNAME",
+            "COHERE_COMPASS_PASSWORD",
+        ]
+        return all(os.getenv(var) is not None for var in vars)
+
+    def __init__(
+            self,
+            compass_url=None,
+            compass_username=None,
+            compass_password=None,
+        ):
+        """Initialize the Compass tool. Pass the Compass URL, username, and password 
+        as arguments or as environment variables."""
+        self.url = compass_url or os.getenv("COHERE_COMPASS_URL")
+        self.username = compass_username or os.getenv("COHERE_COMPASS_USERNAME")
+        self.password = compass_password or os.getenv("COHERE_COMPASS_PASSWORD")
+        self.parser_client = CompassParserClient(
+            parser_url=self.url + "/parse",
+            parser_config=self.username,
+            password=self.password,
```

Comment:
This doesn't specify a `MetadataConfig` anywhere, which mean's were just using the default with no metadata extraction. This is perfectly fine, but just calling out we're accepting the defaults here.

Reply:
I passed the parser_config and metadata_config as parameters with default values. This should allow for customization when the tool is created. Also included the ability to pass parser_config and metadata_config in the to the _parse method.